src/app/game-manager/game-manager.service.ts

```diff
@@ -0,0 +1,46 @@
+import { TimerService } from './../shared/timer/timer.service';
+import { Injectable } from '@angular/core';
+import { Observable, ReplaySubject } from 'rxjs';
+import { DataRepository } from '../repository/repository.service';
+import { Game } from '../shared/models/game.model';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class GameManagerService {
+  private currentGame = new ReplaySubject<Game>();
+  timerTimeout!: ReturnType<typeof setTimeout>;
+  gameWillStartSoon = false;
+  gameIsStarted = false;
+
+  constructor(
+    private repo: DataRepository,
+    private timerService: TimerService,
+  ) {}
+
+  get currentGameObservable(): Observable<Game> {
+    return this.currentGame.asObservable();
+  }
+
+  refreshCurrentGameInfo(): void {
+    // start the timer with calculated values;
```

Comment:
usually comments for whole methods/functions are written as JSDoc format (please check that notation)

Reply:
ok